runtime/ops/tty.rs

```diff
@@ -305,3 +319,28 @@ mod tests {
     );
   }
 }
+
+#[op(fast)]
+pub fn op_read_line_prompt(
+  prompt_text: String,
+  default_value: String,
+) -> Result<Option<String>, AnyError> {
+  let mut editor = Editor::<()>::new().expect("Failed to create editor.");
+
+  editor.set_keyseq_timeout(1);
+  editor
+    .bind_sequence(KeyEvent(KeyCode::Esc, Modifiers::empty()), Cmd::Interrupt);
+
+  let read_result =
+    editor.readline_with_initial(&prompt_text, (&default_value, ""));
+  match read_result {
+    Ok(line) => Ok(Some(line)),
+    Err(ReadlineError::Interrupted | ReadlineError::Eof) => Ok(None),
+    Err(err) => Err(err.into()),
+  }
+}
+
+#[op(fast)]
+pub fn op_use_color() -> bool {
+  colors::use_color()
+}
```

Comment:
This op seems unnecessary. You can reuse some of the code from `ext/console/01_console.js` for color handling. I think that makes more sense than adding another codepath that needs to be kept in sync.